### PR TITLE
enhance: Refactor hybrid search

### PR DESCRIPTION
**Pull request by czs007**
issue: https://github.com/milvus-io/milvus/issues/25639 https://github.com/milvus-io/milvus/issues/31368
pr :https://github.com/milvus-io/milvus/pull/32020

### CODE REVIEW

File: internal/proto/internal.proto [hunk 1]
```diff
@@ -102,18 +116,22 @@ message SearchRequest {
...
-  uint64 mvcc_timestamp = 11;
-  uint64 guarantee_timestamp = 12;
-  uint64 timeout_timestamp = 13;
+message SubSearchResults {
+  string metric_type = 1;
+  int64 num_queries = 2;
+  int64 top_k = 3;
+  // schema.SearchResultsData inside
+  bytes sliced_blob = 4;
+  int64 sliced_num_count = 5;
```
Comment (zhagnlu) [inline internal/proto/internal.proto:131]: what's meaning? sliced_size ?

Comment (czs007) [reply internal/proto/internal.proto:131]: num of slice. it will be always 1.

File: internal/querynodev2/segments/result.go [hunk 2]
```diff
@@ -102,6 +102,88 @@ func ReduceSearchResults(ctx context.Context, results []*internalpb.SearchResult
...
+		}
+	}
+	searchResults := &internalpb.SearchResults{
+		IsAdvanced:   true,
+		ChannelsMvcc: channelsMvcc,
+	}
+
+	for _, result := range results {
+		if !result.GetIsAdvanced() {
+			continue
```
Comment (zhagnlu) [inline internal/querynodev2/segments/result.go:123]: Is that redundant or check assert it?

**Pull request merged**